### PR TITLE
[FIX] web: fix x2many in list mode

### PR DESCRIPTION
**Pull request by FrancoisGe**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/list/list_renderer.js [hunk 1]
```diff
@@ -1201,7 +1200,11 @@ export class ListRenderer extends Component {
 
                 if (futureRecord) {
                     futureRecord.switchMode("edit");
-                } else if (this.lastIsDirty || !record.canBeAbandoned || this.lastCreatingAction) {
+                } else if (
```
Comment (Polymorphe57) [inline addons/web/static/src/views/list/list_renderer.js:1191]: test does not correspond to scenario in commit message

File: addons/web/static/src/views/list/list_renderer.js [hunk 2]
```diff
@@ -1367,20 +1370,6 @@ export class ListRenderer extends Component {
         return false;
     }
 
-    async onCreateAction(context) {
```
Comment (Polymorphe57) [inline addons/web/static/src/views/list/list_renderer.js:1370]: does not seem related to problem

File: addons/web/static/tests/views/fields/one2many_field_tests.js [hunk 3]
```diff
@@ -3785,6 +3785,41 @@ QUnit.module("Fields", (hooks) => {
         assert.hasClass(target.querySelectorAll(".o_data_row")[3], "o_selected_row");
     });
 
+    QUnit.test(
+        "creating a new line in an o2m with a handler does not focus the handler",
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/one2many_field_tests.js:3789]: with a handle field?

File: addons/web/static/tests/views/fields/char_field_tests.js [hunk 4]
```diff
@@ -182,9 +182,7 @@ QUnit.module("Fields", (hooks) => {
         );
```
Comment (Polymorphe57) [inline addons/web/static/tests/views/fields/char_field_tests.js:182]: in the or in some?
the solution to what? you present your commit as a REF. Is it an adaptation required because of some other fix? If yes it would be nice to put the commit before that fix or merge it with it

File: addons/web/static/src/views/list/list_renderer.js [hunk 5]
```diff
@@ -1370,30 +1370,6 @@ export class ListRenderer extends Component {
         return false;
     }
 
-    /**
```
Comment (Polymorphe57) [inline addons/web/static/src/views/list/list_renderer.js:1384]: [REF] web: focus of a X2many

This commit changes the behaviour of a X2many in a
form view when a "tab" or "shift-tab" make it focused.

Before this commit, a new empty row would be created automatically.

After this commit, the button "Add a line" takes focus and no new line is created.
If one wants to add one, that can be done by pressing "Enter".

File: addons/web/static/src/views/relational_model.js [hunk 6]
```diff
@@ -1212,7 +1212,9 @@ export class Record extends DataPoint {
      * @returns {Promise<boolean>}
      */
     async _save(options = { stayInEdition: false, noReload: false }) {
-        this.model.env.bus.trigger("RELATIONAL_MODEL:NEED_LOCAL_CHANGES");
+        const proms = [];
```
Comment (Polymorphe57) [inline addons/web/static/src/views/relational_model.js:1215]: should be done in a fix of domain field no?

File: addons/web/static/src/views/fields/input_field_hook.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 /** @odoo-module **/
```
Comment (Polymorphe57) [inline addons/web/static/src/views/fields/input_field_hook.js:1]: ???? [REF] web: earlier record update for dirty field

A field input always asks for a record update when it is the target of a "change" event
but such an event might occur sometimes before some keyboard events (e.g. a "Enter"). In order to make
keyboard navigation work, it is necessary to make a dirty field to communicate its new value before a change event and make sure the record is correctly updated before being saved.

Before this commit, this was done via a system based on a special event triggered on the model each time a record is
saved but that solution was not appropriate:

- the updates done in that way were not correctly awaited.
- they would be asked for even if the save would not be the result of a keyboard event.

Here we replace the old system by a simpler one that works. When some specific keyboard events occur, a dirty field
ask for an update.
 



Before this commit, when a record would be saved, it would ask to all its related fields
to communicate any input change that would be not known by it, that is ask for an update.
Normally, an update is asked whenever a field input is the targe of a "change" event
Unfortunately, the udpates made that way were not awaited so the record could be
saved without the change.

The purpose of this commit is to not have to tell the fields to apply
their update when saving. The only known way to save a record without
applying updates to the current fields is to apply updates to the keynav.
We will apply an update when we do a tab/shift+tab/enter.

Comment (Polymorphe57) [reply addons/web/static/src/views/fields/input_field_hook.js:1]: occur **after** some keyboard

File: addons/web/static/src/views/fields/input_field_hook.js [hunk 8]
```diff
@@ -134,17 +144,16 @@ export function useInputField(params) {
...
-                component.props.setDirty(isDirty);
-            }
-
             if (isInvalid) {
                 return;
             }
 
             if (val !== component.props.value) {
-                component.props.update(val);
+                await component.props.update(val);
```
Comment (Polymorphe57) [inline addons/web/static/src/views/fields/input_field_hook.js:152]: not nessary anymore

File: addons/web/static/src/views/list/list_renderer.js [hunk 9]
```diff
@@ -959,10 +959,6 @@ export class ListRenderer extends Component {
         const row = cell.parentElement;
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/web/static/src/views/list/list_renderer.js:960]: in ListRenderer

[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/web/static/src/views/list/list_renderer.js:960]: would make
a validitity check

File: addons/web/static/src/views/list/list_renderer.js [hunk 10]
```diff
@@ -1371,44 +1371,6 @@ export class ListRenderer extends Component {
         return false;
```
Comment (Polymorphe57) [inline addons/web/static/src/views/list/list_renderer.js:1359]: make it **be** focused

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo retry

**Comment by aab-odoo** [comment]
robodoo retry

**Comment by aab-odoo** [comment]
robodoo r+
